semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -3056,6 +3057,85 @@ and m_directive a b =
   | G.OtherDirective _ ->
       fail ()
 
+and m_directive_vs_def a b =
+  let f (filea, importsa) = function
+    (* const x = require('y');
+     *
+     * This is for JS but there is no need to gate on language, since the
```

Comment:
imo we might as well gate on JS anyway, it doesn't make the code harder to read and maybe someone will add some language that uses `require` as a CallSpecial but doesn't want this behavior